src/remote/run_jobs.jl

```diff
@@ -0,0 +1,198 @@
+    )
+
+    return circuit_json
+end
+
+
+struct Client 
+    host::String
+    user::String
+    access_token::String
+end
+
+get_host(client::Client)= client.host
+
+# submits circuit to host and returns circuitID
+function submit_circuit(client::Client,circuit_json::String;verbose=false) 
+
+    path_url=joinpath(get_host(client),"circuits")
+    
+    response=HTTP.post(
+        path_url, 
+        headers=Dict(
+            "Authorization"=>"Bearer $(client.access_token)",
+            "Content-Type"=>"application/json"
+            ),
+        body=circuit_json,
+    )
+    
+    formatted_response=Dict(response)
+
+    body=JSON.parse(formatted_response["body"])
+    
+    if verbose
+        printout_response(:submit_circuit,formatted_response,typeof(response))
+    end
+
+    return body["circuitID"]
+end
+
+function printout_response(fname::Symbol,formatted_response::Dict{String,Any},type_response::Type)
+    println("\n#############################################")
+    println("\n\t$fname() returns: \n")
+
+    println("Response type: $type_response")
+
+    for (key,val) in formatted_response
+        println("")
+        println("$key :\t$val")
+    end
+
+    println("")
+end
+
+
+function get_status(client::Client,circuitID::String;verbose=false)
+
+    path_url=joinpath(get_host(client),"circuits/$circuitID")
+    
+    response=HTTP.get(
+        path_url, 
+        headers=Dict(
+            "Authorization"=>"Bearer $(client.access_token)",
+            "Content-Type"=>"application/json"
+            )
+        )
+
+    formatted_response=Dict(response)
+
+    body=JSON.parse(formatted_response["body"])
+    
+    if verbose
+        printout_response(:get_status,formatted_response,typeof(response))
+    end
+
+    return body["status"]["type"]
+end
+
+function get_result(client::Client,circuitID::String;verbose=false)
+
+    path_url=joinpath(get_host(client),"circuits/$circuitID/result")
+    
+    response=HTTP.get(
+        path_url, 
+        headers=Dict(
+            "Authorization"=>"Bearer $(client.access_token)",
+            "Content-Type"=>"application/json"
+            )
+        )
+
+    formatted_response=Dict(response)
+
+    body=JSON.parse(formatted_response["body"])
+    
+    if verbose
+        printout_response(:get_result,formatted_response,typeof(response))
+    end
+
+    return body["histogram"]
+end
+
+
+abstract type Service end
+
+struct QPUService <: Service
+    client::Client
+    qpu::QPU
+
+    function QPUService(client::Client,qpu::QPU)
+        if client.host!=qpu.host
+            throw(
+                ArgumentError(
+                    "Client host $(get_host(client)) and qpu host $(get_host(qpu)) do not match."
+                    )
+                )
+        end
+
+        new(client,qpu)
+    end
+end
+
+get_client(qpu_service::QPUService)=qpu_service.client
+get_qpu(qpu_service::QPUService)=qpu_service.qpu
+
+function Base.Dict(response::HTTP.Messages.Response)
```

Comment:
This is type piracy and could break external code if they import this package

See https://docs.julialang.org/en/v1/manual/style-guide/#Avoid-type-piracy.